components/serializers.py

```diff
@@ -159,3 +195,92 @@ class Meta:
             "catalog",
             "controls_count",
         )
+
+
+class ComponentControlSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Component
+        fields = (
+            "pk",
+            "controls",
+            "description",
+        )
+
+    def update(self, instance, validated_data):
+        # print("controlId:", validated_data.get("control_id"))
+        # if control_id := validated_data.get("control_id"):
+        #     print("controlId:", control_id)
+        print("starting json", instance.component_json)
+        if controls := validated_data.get(
+            "controls"
+        ):  # and controls not in instance.controls:
+            print("controls", controls)
+            print("existing controls", instance.controls)
+            existing_control = controls not in instance.controls
+            print("existing_control", existing_control)
+            # instance.controls.append(controls)
+            if not existing_control:
+                instance.controls = list(set(instance.controls).union(controls))
+
+            print(controls)
+            if description := validated_data.get("description"):
```

Comment:
I nested the description inside of the controls because we need the controls to get the id we are changing the description for... might be able to combined them :/

Reply:
I think this might be easier once #84 is in, but I think it would be better to raise an error here if `control_id` is required, but isn't present. 

Does beg the question though: what is `instance` in this case? A project?

Reply:
`instance` is a component